test/e2e/run-all.js

```diff
@@ -212,12 +213,26 @@ async function main() {
 
   console.log('My test list:', myTestList);
 
+  const changedOrNewTests = await fetchChangedE2eFiles();
+  console.log('Spec files that will be re-run:', changedOrNewTests);
```

Comment:
this is left for debugging purposes

Reply:
You're going to sacrifice some parallel execution speed by doing it this way.  It also only has to be done if it's running on CircleCI, and not locally.

It would be a better approach if at the top of `function runningOnCircleCI()`, you looked at `testPaths` and then duplicated 5 times each thing that was also in `changedOrNewTests`.

This would allow it to distribute the workload evenly across the VMs.

Reply:
thank you for your suggestion @HowardBraham That's definetly a good point! I was thinking it would be negligible given the small amount of retries, but it's true that we can benefit from parallelization witha small tweak. I'll add the changes :bowing_woman:

Reply:
changes added :+1: